### PR TITLE
Introduce DistributedTransactionAdmin

### PR DESCRIPTION
**Pull request by brfrn169**
This PR introduces a new `DistributedTransactionAdmin` class that abstracts the administrative operations for distributed transactions like createNamespace/createTable/getTableMetadata. This PR also adds the implementation classes of `DistributedTransactionAdmin`, `ConsensusCommitAdmin` and `JdbcTransactionAdmin`. I will add inline comment for more details. Please take a look!

### CODE REVIEW

File: core/src/main/java/com/scalar/db/api/DistributedTransactionAdmin.java [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+import com.scalar.db.exception.storage.ExecutionException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * An administrative interface for distributed transaction implementations. The user can execute
+ * administrative operations with it like createNamespace/createTable/getTableMetadata.
+ */
+public interface DistributedTransactionAdmin {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/api/DistributedTransactionAdmin.java:12]: This is a new API class, `DistributedTransactionAdmin`. With it, we can execute administrative operations for distributed transaction implementations.

File: core/src/main/java/com/scalar/db/api/DistributedTransactionAdmin.java [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+   */
+  void truncateCoordinatorTable() throws ExecutionException;
+
+  /**
+   * Returns true if a coordinator table exists.
+   *
+   * @return true if a coordinator table exists, false otherwise
+   * @throws ExecutionException if the operation failed
+   */
+  boolean coordinatorTableExists() throws ExecutionException;
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/api/DistributedTransactionAdmin.java:383]: This class also has the methods to operate a coordinator table.

File: core/src/main/java/com/scalar/db/service/TransactionFactory.java [hunk 3]
```diff
@@ -26,6 +27,15 @@ public DistributedTransactionManager getTransactionManager() {
     return injector.getInstance(DistributedTransactionManager.class);
   }
 
+  /**
+   * Returns a {@link DistributedTransactionAdmin} instance
+   *
+   * @return a {@link DistributedTransactionAdmin} instance
+   */
+  public DistributedTransactionAdmin getTransactionAdmin() {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/service/TransactionFactory.java:35]: We can get an instance of `DistributedTransactionAdmin` with this `TransactionFactory.getTransactionAdmin()`.

File: core/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitAdmin.java [hunk 4]
```diff
@@ -1,146 +1,111 @@
...
 import com.scalar.db.api.TableMetadata;
 import com.scalar.db.exception.storage.ExecutionException;
-import java.util.Collections;
 import java.util.Map;
+import java.util.Set;
 import javax.annotation.concurrent.ThreadSafe;
 
 @ThreadSafe
-public class ConsensusCommitAdmin {
+public class ConsensusCommitAdmin implements DistributedTransactionAdmin {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitAdmin.java:16]: `ConsensusCommitAdmin` is one of implementations of `DistributedTransactionAdmin`. This class abstracts the Consensus Commit things (like transaction metadata).

File: core/src/main/java/com/scalar/db/transaction/jdbc/JdbcTransactionAdmin.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+import com.scalar.db.api.TableMetadata;
+import com.scalar.db.exception.storage.ExecutionException;
+import com.scalar.db.storage.jdbc.JdbcAdmin;
+import com.scalar.db.storage.jdbc.JdbcConfig;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.concurrent.ThreadSafe;
+
+@ThreadSafe
+public class JdbcTransactionAdmin implements DistributedTransactionAdmin {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/transaction/jdbc/JdbcTransactionAdmin.java:15]: `JdbcTransactionAdmin` is the other one of the implementations of  `DistributedTransactionAdmin`. This class just transfers the operations to `JdbcAdmin` actually.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**